tools/mksurfdata_map/src/mklaiMod.F90

```diff
@@ -247,11 +249,11 @@ integer,  pointer     :: laimask(:,:)     ! lai+sai output mask for each plant f
      ! Loop over pft types to do mapping
 
      do l = 0, numpft_i - 1
-        mask_src(:) = 1._r8 
-        call gridmap_areaave(tgridmap, mlai_i(:,l) , mlai_o(:,l) , nodata=0._r8, mask_src=mask_src)
-        call gridmap_areaave(tgridmap, msai_i(:,l) , msai_o(:,l) , nodata=0._r8, mask_src=mask_src)
-        call gridmap_areaave(tgridmap, mhgtt_i(:,l), mhgtt_o(:,l), nodata=0._r8, mask_src=mask_src)
-        call gridmap_areaave(tgridmap, mhgtb_i(:,l), mhgtb_o(:,l), nodata=0._r8, mask_src=mask_src)
+        mask_src(:) = tdomain%mask(:)
```

Comment:
This assignment should be taken outside the loop. But is it even necessary? Why not just use tdomain%mask directly rather than using this copy?

Reply:
Moving the assignment outside the loop makes sense (DONE).

However, I get the following error in mkpftMod.F90 and mklaiMod.F90 when using tdomain%mask directly:

`mkpftMod.F90(516): error #6285: There is no matching specific subroutine for this generic subroutine call.   [GRIDMAP_AREAAVE]`
`call gridmap_areaave(tgridmap, pctnatveg_i, pctnatveg_o, nodata=0._r8, mask_src=tdomain%mask, wtnorm=wtnorm)`
`-------------^`

Reply:
Ah, I see - because tdomain%mask is an integer, while the subroutine expects a real(r8). Since masks are, by definition, 0/1 (right?) it seems best to change the subroutine to expect an integer mask rather than a real-valued mask, at which point you can pass in tdomain%mask directly. Does that seem reasonable, or do you think that won't work well for some reason?

Reply:
Sounds reasonable to me.